docs/requirements.txt

```diff
@@ -5,7 +5,7 @@
 breathe
 ipython
 numpydoc
-pydata-sphinx-theme~=0.14
+pydata-sphinx-theme~=0.15.2
```

Comment:
@jorisvandenbossche @AlenkaF Do we need `.2` with `~=`?

Reply:
AFAIK if `~=` is added then pip will install the latest `0.15.x` version that is higher or equal to `0.15.2.` So I think this is fine.

Reply:
Then why are the checks failing?

Reply:
I do not think they are connected.
We can still try and change to `==0.15.*` to see if it makes a difference.

Reply:
Or remove the `.2` as Kou suggested 😊

Reply:
![image](https://github.com/apache/arrow/assets/146909065/aa8b43fe-6472-43e4-afbf-ce239e319a28)
Did this happen during #39658 if so how was it resolved? @AlenkaF  @kou

Reply:
>  Could you please tell if this happened during #39658 if so how was it resolved?

This is a failure on a Python 3.8 build because the theme package requires newer version of Python:

```
#8 10.21 └─ pydata-sphinx-theme 0.15.2  is installable and it requires
#8 10.21    └─ python >=3.9  with the potential options
```

That means that on the specific `Python / AMD64 Conda Python 3.8 Pandas 1.0` CI build the upper pin for the `pydata-sphinx-theme` will have to stay as is (`0.14.*`). @kou maybe you can help to find correct dockerfiles to correct?

> Also while building the documentation I am getting this error could you please tell me how to resolve this.I am following(https://arrow.apache.org/docs/developers/documentation.html#building)

Could you check you have all dependencies installed? https://github.com/apache/arrow/blob/22f2cfd1e1ebe49016b6d97c49f494287a98d02f/docs/requirements.txt#L5

Reply:
Could you try this?

```diff
diff --git a/ci/docker/conda-python-docs.dockerfile b/ci/docker/conda-python-docs.dockerfile
new file mode 100644
index 0000000000..0cee0f5b49
--- /dev/null
+++ b/ci/docker/conda-python-docs.dockerfile
@@ -0,0 +1,26 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG repo
+ARG arch=amd64
+ARG python=3.8
+ARG pandas=latest
+FROM ${repo}:${arch}-conda-python-${python}-pandas-${pandas}
+
+COPY ci/conda_env_sphinx.txt /arrow/ci/
+RUN mamba install -q -y --file arrow/ci/conda_env_sphinx.txt && \
+    mamba clean --all
diff --git a/ci/docker/conda-python-pandas.dockerfile b/ci/docker/conda-python-pandas.dockerfile
index 83ad52a13d..dc29fb50bd 100644
--- a/ci/docker/conda-python-pandas.dockerfile
+++ b/ci/docker/conda-python-pandas.dockerfile
@@ -23,10 +23,8 @@ FROM ${repo}:${arch}-conda-python-${python}
 ARG pandas=latest
 ARG numpy=latest
 
-# the doc builds are using the conda-python-pandas image,
-# so ensure to install doc requirements
-COPY ci/conda_env_sphinx.txt /arrow/ci/
-RUN mamba install -q -y --file arrow/ci/conda_env_sphinx.txt && \
+COPY ci/conda_env_python.txt /arrow/ci/
+RUN mamba install -q -y --file arrow/ci/conda_env_python.txt && \
     mamba clean --all
 
 COPY ci/scripts/install_pandas.sh /arrow/ci/scripts/
diff --git a/docker-compose.yml b/docker-compose.yml
index 47e74d088f..0f7d7f6c4a 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1196,7 +1196,16 @@ services:
     # Only a single rule is enabled for now to check undocumented arguments.
     # We should extend the list of enabled rules after adding this build to
     # the CI pipeline.
-    image: ${REPO}:${ARCH}-conda-python-${PYTHON}-pandas-${PANDAS}
+    image: ${REPO}:${ARCH}-conda-python-${PYTHON}-docs
+      context: .
+      dockerfile: ci/docker/conda-python-docs.dockerfile
+      cache_from:
+        - ${REPO}:${ARCH}-conda-python-${PYTHON}-docs
+      args:
+        repo: ${REPO}
+        arch: ${ARCH}
+        python: ${PYTHON}
+        pandas: ${PANDAS}
     environment:
       <<: [*common, *ccache]
       ARROW_SUBSTRAIT: "ON"
```

Reply:
@AlenkaF I building the docs using these sets of cmd
```
pushd arrow/docs
make cpp
popd
``` 
I checked and it seems breathe is installed but I am still getting the error
```powershell
PS C:\Users\Divyansh\Documents\GitHub\arrow\docs> pip install breathe 
DEPRECATION: Loading egg at c:\users\divyansh\appdata\roaming\python\python312\site-packages\pyarrow-14.0.0.dev331+g4bf777a6a-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\divyansh\appdata\roaming\python\python312\site-packages\pyarrow-16.0.0.dev45+g9f0c727d9-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\python312\lib\site-packages\pyarrow-14.0.0.dev331+g4bf777a6a-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\python312\lib\site-packages\pyarrow-16.0.0.dev45+g9f0c727d9-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
Requirement already satisfied: breathe in c:\python312\lib\site-packages (4.35.0)
Requirement already satisfied: Sphinx!=5.0.0,>=4.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from breathe) (6.2.0)
Requirement already satisfied: docutils>=0.12 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from breathe) (0.19)
Requirement already satisfied: sphinxcontrib-applehelp in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.0.8)
Requirement already satisfied: sphinxcontrib-devhelp in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.0.6)
Requirement already satisfied: sphinxcontrib-jsmath in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.0.1)
Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.0 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (2.0.5)
Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.5 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.1.10)
Requirement already satisfied: sphinxcontrib-qthelp in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.0.7)
Requirement already satisfied: Jinja2>=3.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (3.1.3)
Requirement already satisfied: Pygments>=2.13 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (2.17.2)
Requirement already satisfied: snowballstemmer>=2.0 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (2.2.0)
Requirement already satisfied: babel>=2.9 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (2.14.0)
Requirement already satisfied: alabaster<0.8,>=0.7 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (0.7.16)
Requirement already satisfied: imagesize>=1.3 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (1.4.1)
Requirement already satisfied: requests>=2.25.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (2.31.0)
Requirement already satisfied: packaging>=21.0 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (23.2)
Requirement already satisfied: colorama>=0.4.5 in c:\python312\lib\site-packages (from Sphinx!=5.0.0,>=4.0->breathe) (0.4.6)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Jinja2>=3.0->Sphinx!=5.0.0,>=4.0->breathe) (2.1.4)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->Sphinx!=5.0.0,>=4.0->breathe) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->Sphinx!=5.0.0,>=4.0->breathe) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\python312\lib\site-packages (from requests>=2.25.0->Sphinx!=5.0.0,>=4.0->breathe) (2.2.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->Sphinx!=5.0.0,>=4.0->breathe) (2023.11.17)
PS C:\Users\Divyansh\Documents\GitHub\arrow\docs> make cpp
sphinx-build -b html -j8 -c source source/cpp _build/html/cpp
Running Sphinx v5.0.2

Extension error:
Could not import extension breathe (exception: No module named 'breathe')
make: *** [Makefile:141: cpp] Error 2

```

Reply:
> > Could you please tell if this happened during #39658 if so how was it resolved?
> 
> This is a failure on a Python 3.8 build because the theme package requires newer version of Python:
> 
> ```
> #8 10.21 └─ pydata-sphinx-theme 0.15.2  is installable and it requires
> #8 10.21    └─ python >=3.9  with the potential options
> ```
> 
> That means that on the specific `Python / AMD64 Conda Python 3.8 Pandas 1.0` CI build the upper pin for the `pydata-sphinx-theme` will have to stay as is (`0.14.*`). @kou maybe you can help to find correct dockerfiles to correct?
> 
> > Also while building the documentation I am getting this error could you please tell me how to resolve this.I am following(https://arrow.apache.org/docs/developers/documentation.html#building)
> 
> Could you check you have all dependencies installed?
> 
> https://github.com/apache/arrow/blob/22f2cfd1e1ebe49016b6d97c49f494287a98d02f/docs/requirements.txt#L5

Sorry for the trouble but breathe dependency is already installed @AlenkaF 
```powershell
PS C:\Users\Divyansh\Documents\GitHub> pip install -r arrow/docs/requirements.txt
Requirement already satisfied: breathe in c:\python312\lib\site-packages (from -r arrow/docs/requirements.txt (line 5)) (4.35.0)
```

Reply:
Hm, not sure what is the issue then. Could you try a clean build and add full log in case it still errors?

Reply:
Please take a look @AlenkaF 

```powershell
PS C:\Users\Divyansh\Documents\GitHub> pip install -r arrow/docs/requirements.txt
DEPRECATION: Loading egg at c:\users\divyansh\appdata\roaming\python\python312\site-packages\pyarrow-14.0.0.dev331+g4bf777a6a-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\divyansh\appdata\roaming\python\python312\site-packages\pyarrow-16.0.0.dev45+g9f0c727d9-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\python312\lib\site-packages\pyarrow-14.0.0.dev331+g4bf777a6a-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\python312\lib\site-packages\pyarrow-16.0.0.dev45+g9f0c727d9-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\python312\lib\site-packages\pyarrow-16.0.0.dev49+g102094668.d20240206-py3.12-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
Requirement already satisfied: breathe in c:\python312\lib\site-packages (from -r arrow/docs/requirements.txt (line 5)) (4.35.0)
Requirement already satisfied: ipython in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 6)) (8.21.0)
Requirement already satisfied: numpydoc in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 7)) (1.1.0)
Requirement already satisfied: pydata-sphinx-theme~=0.15 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 8)) (0.15.2)
Requirement already satisfied: sphinx-autobuild in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 9)) (2021.3.14)      
Requirement already satisfied: sphinx-design in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 10)) (0.5.0)
Requirement already satisfied: sphinx-copybutton in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 11)) (0.5.2)        
Requirement already satisfied: sphinx==6.2 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 12)) (6.2.0)
Requirement already satisfied: pandas in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from -r arrow/docs/requirements.txt (line 13)) (2.2.0)
Requirement already satisfied: sphinxcontrib-applehelp in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.0.8)
Requirement already satisfied: sphinxcontrib-devhelp in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.0.6)
Requirement already satisfied: sphinxcontrib-jsmath in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.0.1)
Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.0 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.0.5)
Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.5 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.1.10)
Requirement already satisfied: sphinxcontrib-qthelp in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.0.7)
Requirement already satisfied: Jinja2>=3.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (3.1.3) 
Requirement already satisfied: Pygments>=2.13 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.17.2)
Requirement already satisfied: docutils<0.20,>=0.18.1 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (0.19)
Requirement already satisfied: snowballstemmer>=2.0 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.2.0)
Requirement already satisfied: babel>=2.9 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.14.0) 
Requirement already satisfied: alabaster<0.8,>=0.7 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (0.7.16)
Requirement already satisfied: imagesize>=1.3 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (1.4.1)
Requirement already satisfied: requests>=2.25.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.31.0)
Requirement already satisfied: packaging>=21.0 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (23.2)
Requirement already satisfied: colorama>=0.4.5 in c:\python312\lib\site-packages (from sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (0.4.6)
Requirement already satisfied: decorator in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (5.1.1)
Requirement already satisfied: jedi>=0.16 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (0.19.1)      
Requirement already satisfied: matplotlib-inline in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (0.1.6)
Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.41 in c:\python312\lib\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (3.0.43)
Requirement already satisfied: stack-data in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (0.6.3)       
Requirement already satisfied: traitlets>=5 in c:\python312\lib\site-packages (from ipython->-r arrow/docs/requirements.txt (line 6)) (5.14.1)
Requirement already satisfied: beautifulsoup4 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from pydata-sphinx-theme~=0.15->-r arrow/docs/requirements.txt (line 8)) (4.12.3)
Requirement already satisfied: accessible-pygments in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from pydata-sphinx-theme~=0.15->-r arrow/docs/requirements.txt (line 8)) (0.0.4)
Requirement already satisfied: typing-extensions in c:\python312\lib\site-packages (from pydata-sphinx-theme~=0.15->-r arrow/docs/requirements.txt (line 8)) (4.9.0)
Requirement already satisfied: livereload in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from sphinx-autobuild->-r arrow/docs/requirements.txt (line 9)) (2.6.3)
Requirement already satisfied: numpy<2,>=1.26.0 in c:\python312\lib\site-packages (from pandas->-r arrow/docs/requirements.txt (line 13)) (1.26.2)
Requirement already satisfied: python-dateutil>=2.8.2 in c:\python312\lib\site-packages (from pandas->-r arrow/docs/requirements.txt (line 13)) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in c:\python312\lib\site-packages (from pandas->-r arrow/docs/requirements.txt (line 13)) (2023.4)
Requirement already satisfied: tzdata>=2022.7 in c:\python312\lib\site-packages (from pandas->-r arrow/docs/requirements.txt (line 13)) (2023.4)
Requirement already satisfied: parso<0.9.0,>=0.8.3 in c:\python312\lib\site-packages (from jedi>=0.16->ipython->-r arrow/docs/requirements.txt (line 6)) (0.8.3)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from Jinja2>=3.0->sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.1.4)
Requirement already satisfied: wcwidth in c:\python312\lib\site-packages (from prompt-toolkit<3.1.0,>=3.0.41->ipython->-r arrow/docs/requirements.txt (line 6)) (0.2.13)
Requirement already satisfied: six>=1.5 in c:\python312\lib\site-packages (from python-dateutil>=2.8.2->pandas->-r arrow/docs/requirements.txt (line 13)) (1.16.0)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\python312\lib\site-packages (from requests>=2.25.0->sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2.2.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from requests>=2.25.0->sphinx==6.2->-r arrow/docs/requirements.txt (line 12)) (2023.11.17)
Requirement already satisfied: soupsieve>1.2 in c:\python312\lib\site-packages (from beautifulsoup4->pydata-sphinx-theme~=0.15->-r arrow/docs/requirements.txt (line 8)) (2.5)
Requirement already satisfied: tornado in c:\python312\lib\site-packages (from livereload->sphinx-autobuild->-r arrow/docs/requirements.txt (line 9)) (6.4)
Requirement already satisfied: executing>=1.2.0 in c:\python312\lib\site-packages (from stack-data->ipython->-r arrow/docs/requirements.txt (line 6)) (2.0.1)
Requirement already satisfied: asttokens>=2.1.0 in c:\users\divyansh\appdata\roaming\python\python312\site-packages (from stack-data->ipython->-r arrow/docs/requirements.txt (line 6)) (2.4.1)
Requirement already satisfied: pure-eval in c:\python312\lib\site-packages (from stack-data->ipython->-r arrow/docs/requirements.txt (line 6)) (0.2.2)

[notice] A new release of pip is available: 23.3.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
PS C:\Users\Divyansh\Documents\GitHub> pushd arrow/docs
PS C:\Users\Divyansh\Documents\GitHub\arrow\docs> make cpp
sphinx-build -b html -j8 -c source source/cpp _build/html/cpp
Running Sphinx v5.0.2

Extension error:
Could not import extension breathe (exception: No module named 'breathe')
make: *** [Makefile:141: cpp] Error 2
PS C:\Users\Divyansh\Documents\GitHub\arrow\docs> popd


```

Reply:
The one thing I notice is that the location where the packages are installed differs. Some are installed in `c:\python312\lib\site-packages\...` (for example `breathe`) while others are in `c:\users\divyansh\appdata\roaming\python\python312\site-packages` (for example `Sphinx`).

I would suggest creating a clean build meaning a new clean python environment where you install the packages again, making sure they are in the same location.

Reply:
@AlenkaF Thank you it worked. Also while I was making the changes suggested by @kou about adding the docker image this is coming

![image](https://github.com/apache/arrow/assets/146909065/59ed44f0-202d-4b7f-bf62-6a421e99522d)

when i used `yamllint docker-compose.yml` cmd this erorr is coming

```powershell
  1201:21   error    syntax error: mapping values are not allowed here (syntax)
```

Reply:
Not sure but looking at similar syntax here: https://github.com/apache/arrow/blob/bbe59b35de33a0534fc76c9617aa4746031ce16c/docker-compose.yml#L215-L243 I would first try adding `build:` command and looking for other differences.

Reply:
It seems that you have too much indent for line 1201-1208.

Anyway, we should do it ( https://github.com/apache/arrow/pull/39879#discussion_r1477187035 ) in a separated PR as @AlenkaF suggested ( https://github.com/apache/arrow/pull/39879#issuecomment-1926636979 ).

Could you open an issue for it to work on it as a separated task?

Reply:
> If you want @kou I can work on the indentation in this pr ?
> BTW I think these are the errors coming in the `docker-compose.yml` file so if I am supposed to open a separate issue should I work on only this part where I made the code changes

@kou could you please take a look at the quoted text

Reply:
Please don't do it in this PR. If you want to work on it, please open a new PR for it. It's not related to this PR.

BTW, here is an `yamllint` output on my environment:

```console
$ yamllint docker-compose.yml
docker-compose.yml
  1:1       warning  missing document start "---"  (document-start)
```

It seems that you broke `docker-compose.yml` on your environment.

Reply:
So is this PR good to merge?